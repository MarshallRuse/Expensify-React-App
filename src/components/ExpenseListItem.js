import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = (props) => (
    <Link className='list__item' to={`/edit/${props.id}`}>
        <div>
            <h2 className='list__item__title'>{props.description}</h2>
            <span className='list__item__subtitle'>
                {moment(props.createdAt).format('MMMM Do, YYYY')}
            </span>
        </div>
        <div>
            <h3 className='list__item__amount'>
                {numeral(props.amount / 100).format('$0,0.00')}
            </h3>
        </div>
    </Link>
);

export default ExpenseListItem;

