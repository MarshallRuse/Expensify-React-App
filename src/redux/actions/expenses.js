 import uuid from 'uuid';
 
 // ADD_EXPENSE
export const addExpense = ({
    id = undefined,
    description = '',
    notes = '',
    amount = 0, 
    createdAt = 0
} = {}) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: id ? id : uuid(), // note this works here, not in the arguments
        description,
        notes,
        amount,
        createdAt
    }
});

    // REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
})

    // EDIT EXPENSE
export const editExpense = (id = '', updates = {}) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
})