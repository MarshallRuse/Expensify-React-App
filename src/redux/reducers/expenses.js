const expensesReducerDefaultState = [];

const expensesReducer = ( state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            // .concat() doesn't mutate state directly like .push() would,
            // returns a new array
            return state.concat(action.expense);
        case 'REMOVE_EXPENSE':
            return state.filter(({ id }) => id !== action.id);
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    return { ...expense, ...action.updates};
                } else {
                    return expense;
                }
            })
        case 'SET_EXPENSES':
            return action.expenses;
        default:
            return state;
    }
}

export default expensesReducer;