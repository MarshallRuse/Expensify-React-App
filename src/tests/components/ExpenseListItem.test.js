import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('ExpenseListItem should render correctly', () => {
    const expense = expenses[1];
    const wrapper = shallow(<ExpenseListItem {...expense}/>)
    expect(wrapper).toMatchSnapshot();
})
