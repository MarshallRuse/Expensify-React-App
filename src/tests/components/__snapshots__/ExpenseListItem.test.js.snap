// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseListItem should render correctly 1`] = `
<Link
  className="list__item"
  to="/edit/2"
>
  <div>
    <h2
      className="list__item__title"
    >
      Rent
    </h2>
    <span
      className="list__item__subtitle"
    >
      December 27th, 1969
    </span>
  </div>
  <div>
    <h3
      className="list__item__amount"
    >
      $1,095.00
    </h3>
  </div>
</Link>
`;
